server/db/repositories.js

```diff
@@ -13,8 +14,39 @@ const repositorySchema = new Schema({
     created_at: Date,
     updated_at: Date,
     language: String,
-});
+    stargazersLogin: [String],
+}, { timestamps: { updatedAt: 'recordUpdated_at', createdAt: 'recordCreated_at' } });
 
+repositorySchema.statics.updateRepoWithStargazer = function (repo, stargazer) {
+    const Repository = this;
+
+    return new Promise((resolve, reject) => {
+        Repository.findOne({ githubId: repo.githubId })
+                .then((repository) => {
+                    if (repository) {
+                        if (repository.stargazersLogin.indexOf(stargazer) === -1) {
+                            repository.stargazersLogin.push(stargazer);
+                        }
+                        Object.assign(repository, repo);
```

Comment:
You should better do `Object.assign({}, repository, repo)` to avoid any undesired mutation.